tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,174 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
+#define PLUGIN_NAME "s3_plugin"
+#define TEST_DATA "I love Fluent Bit"
+#define KEY_1 "key1"
+#define KEY_2 "key2"
+#define KEY_3 "key3"
+
+static void check_chunk(struct flb_local_chunk *chunk, char *tag, char *data)
+{
+    char *buffered_data = NULL;
+    size_t buffer_size;
+    int ret;
+    /* Ensure data retreived is same as that which was stored. */
+    if (chunk->tag == tag){
```

Comment:
This should be TEST_CHECK(strcmp(chunk->tag, tag) == 0) I think, instead of an if block

Reply:
Yeah, I agree.